java/build.gradle

```
@@ -144,10 +150,22 @@ dependencies {
     )
 }
 
-// Use -PdryRunFailSkip in the command line to avoid failing the project during an OpenRewrite dryRun.
-task rewriteDryRunFailable {
-    if (!project.hasProperty("dryRunFailSkip")) {
+// Use -PenableDryRunFail in the command line to enable failing the project during any operation with rewriteDryRun.
+// You need to use -PenableCops to be able to invoke the checkstyle, rewriteRun, and rewriteDryRun tasks.
+// The tasks still need to be manually called if nothing else invokes them.
+tasks.named('rewriteDryRun') {
+    task ->
+        if (!project.hasProperty("enableCops")) {
+            task.enabled = false
+        }
+    if (project.hasProperty("enableDryRunFail")) {
         rewrite.failOnDryRunResults = true
     }
-    finalizedBy rewriteDryRun
+}
+
+tasks.named('rewriteRun') {
+    task ->
+        if (!project.hasProperty("enableCops")) {
+            task.enabled = false
+        }
```

You can simplify it as:

```suggestion
        task.enabled = project.hasProperty("enableCops")
```

But I would prefer to get rid of this option, unless it makes the build too slow.